pkg/search/postgres/store.go

```diff
@@ -4,44 +4,62 @@ import (
 	"context"
 	"time"
 
+	"github.com/hashicorp/go-multierror"
 	"github.com/jackc/pgx/v4"
 	"github.com/pkg/errors"
 	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
 	"github.com/stackrox/rox/pkg/auth/permissions"
 	ops "github.com/stackrox/rox/pkg/metrics"
 	"github.com/stackrox/rox/pkg/postgres"
 	"github.com/stackrox/rox/pkg/postgres/pgutils"
 	"github.com/stackrox/rox/pkg/postgres/walker"
 	"github.com/stackrox/rox/pkg/sac"
 	"github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/pkg/utils"
 )
 
 const (
+	batchAfter      = 100
 	cursorBatchSize = 50
 	deleteBatchSize = 5000
 )
 
+// Deleter is an interface that allow deletions of multiple identifiers
+type Deleter interface {
+	DeleteMany(ctx context.Context, identifiers []string) error
+}
+
 // PermissionChecker is a permission checker that could be used by GenericStore
 type PermissionChecker interface {
 	CountAllowed(ctx context.Context) (bool, error)
 	DeleteAllowed(ctx context.Context, keys ...sac.ScopeKey) (bool, error)
 	DeleteManyAllowed(ctx context.Context, keys ...sac.ScopeKey) (bool, error)
```

Comment:
Follow-up candidate:
- As the `ByQuery` and `Many` variants were removed for `Get` and `Upsert`, it might be good to remove `DeleteManyAllowed` from the interface and replace the associated calls with calls to `DeleteAllowed`.
- The simplification of the `PermissionChecker` interface should be reflected in the generated interface template as well as in the implementations (`clusterinit`, `networkgraph/entity` and `vulnerabilityrequest`).

Reply:
ROX-11167

Reply:
- #6907